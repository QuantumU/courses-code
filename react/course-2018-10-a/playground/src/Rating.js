import React, { PureComponent } from 'react';

class Rating extends PureComponent {
  render() {
    return (
      <div>
        {[1, 2, 3, 4, 5].map(index => (
          <span
            onClick={() => {
              this.props.ratingChange(this.props.index, index);
            }}
          >
            {this.props.stars >= index ? '★' : '☆'}
          </span>
        ))}
      </div>
    );
  }
}

export default Rating;
