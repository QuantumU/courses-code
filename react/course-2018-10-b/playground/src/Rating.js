import React, { PureComponent } from 'react';

class Rating extends PureComponent {
  render() {
    const starData = [];
    for (let i = 1; i <= 5; i++) {
      starData.push({
        id: i,
        active: i <= this.props.stars
      });
    }
    return (
      <div>
        {starData.map(data => (
          <span
            key={data.id.toString()}
            className="star"
            onClick={() => {
              this.props.onStarsChange(data.id);
            }}
          >
            {data.active ? '★' : '☆'}
          </span>
        ))}
      </div>
    );
  }
}

export default Rating;
