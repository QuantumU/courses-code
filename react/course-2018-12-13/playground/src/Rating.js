import React, { PureComponent } from 'react';

class Rating extends PureComponent {
  render() {
    let starData = [];
    for (let i = 1; i <= 5; i++) {
      starData.push({ id: i, active: i <= this.props.stars });
    }

    return (
      <div style={{ fontSize: 35, color: 'gold' }}>
        {starData.map(star => (
          <span
            onClick={() => {
              this.props.onRatingChange(star.id);
            }}
          >
            {star.active ? '\u2605' : '\u2606'}
          </span>
        ))}
      </div>
    );
  }
}

export default Rating;
