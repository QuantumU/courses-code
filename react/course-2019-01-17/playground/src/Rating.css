.rating {
  font-size: 30px;
  color: gold;
}