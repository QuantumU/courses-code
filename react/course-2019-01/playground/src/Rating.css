.rating {
  font-size: 25px;
}
