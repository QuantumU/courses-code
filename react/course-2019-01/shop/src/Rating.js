import React from "react";

class Rating extends React.PureComponent {
  /*
    props:
      stars: Number
      interactive: boolean
    events:
      onStarsChange: function(Number)
   */
  render() {
    let starData = [];
    for (let i = 0; i <= 4; i++) {
      starData.push({ id: i+1, active: i+1 <= this.props.stars });
    }

    return (
      <div className="rating">
        {starData.map(star => (
          <span
            style={{ color: star.active ? "red" : "grey" }}
            onClick={clickEvent => {
              if (this.props.interactive) {
                this.props.onStarsChange(star.id);
              }
            }}
            key={star.id.toString()}
          >
            â˜†
          </span>
        ))}
      </div>
    );
  }
}

export default Rating;
