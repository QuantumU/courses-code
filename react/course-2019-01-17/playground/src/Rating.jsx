import React, { Component } from 'react';
import './Rating.css';

class Rating extends Component {
  /*
    props:
      stars: number
    events:
      onStarsChange: function(number)
   */
  render() {
    let starData = [];
    for (let i = 0; i <= 4; i++) {
      starData.push({ id: i, active: i < this.props.stars });
    }

    return (
      <div className="rating">
        {starData.map(star => (
          <span
            key={star.id.toString()}
            onClick={() => {
              this.props.onStarsChange(star.id + 1);
            }}
          >
            {star.active ? '★' : '☆'}
          </span>
        ))}
      </div>
    );
  }
}

export default Rating;
