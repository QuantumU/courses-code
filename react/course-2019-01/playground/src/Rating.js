import React from "react";
import "./Rating.css";

class Rating extends React.Component {
  /*
    props:
      stars: Number
      interactive: boolean
    events:
      onStarsChange: function(Number)
   */
  render() {
    let starData = [];
    for (let i = 1; i <= 5; i++) {
      starData.push({ id: i, active: i <= this.props.stars });
    }

    return (
      <div className="rating">
        {starData.map(star => (
          <span
            style={{ color: star.active ? "red" : "grey" }}
            onClick={clickEvent => {
              if (this.props.interactive) {
                this.props.onStarsChange(star.id);
              }
            }}
          >
            â˜…
          </span>
        ))}
      </div>
    );
  }
}

export default Rating;
