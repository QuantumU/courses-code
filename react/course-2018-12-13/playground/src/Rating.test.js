import React from 'react';
import Rating from './Rating';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Rating stars={4} />).toJSON();
  expect(tree).toMatchSnapshot();
});
