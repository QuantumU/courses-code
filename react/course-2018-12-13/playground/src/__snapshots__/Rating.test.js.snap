// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "color": "gold",
      "fontSize": 35,
    }
  }
>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ☆
  </span>
</div>
`;
